### ISSUE TITLE
java.lang.ClassCastException on device.connect() - version 0.13.0 specific

### ISSUE THREAD

**Issue by vin-lz** [boskokg/flutter_blue#15]
Hi! I'm experiencing an issue when I call `device.connect()` in version 0.13.0. Please have a look at the log:
```
I/BluetoothAdapter(24348): getBluetoothLeScanner
D/BluetoothAdapter(24348): isLeEnabled(): ON
D/BluetoothLeScanner(24348): onScannerRegistered() - status=0 scannerId=7 mScannerId=0
I/zygote64(24348): Do partial code cache collection, code=25KB, data=27KB
I/zygote64(24348): After code cache collection, code=25KB, data=27KB
I/zygote64(24348): Increasing code cache capacity to 128KB
I/flutter (24348):    INFO [💡  21:32:04 FlutterBluePageController]: Found Device
I/BluetoothAdapter(24348): getBluetoothLeScanner
D/BluetoothAdapter(24348): isLeEnabled(): ON
D/BluetoothManager(24348): getConnectedDevices
D/BluetoothGatt(24348): connect() - device: F2:9A:3E:2B:6E:5D, auto: true
D/BluetoothGatt(24348): registerApp()
D/BluetoothGatt(24348): registerApp() - UUID=43b530ba-380a-4784-9b85-de82b101bcd5
D/BluetoothGatt(24348): onClientRegistered() - status=0 clientIf=7
E/MethodChannel#plugins.pauldemarco.com/flutter_blue/methods(24348): Failed to handle method call
E/MethodChannel#plugins.pauldemarco.com/flutter_blue/methods(24348): java.lang.ClassCastException: byte[] cannot be cast to java.lang.String
E/MethodChannel#plugins.pauldemarco.com/flutter_blue/methods(24348): 	at com.pauldemarco.flutter_blue.FlutterBluePlugin.onMethodCall(FlutterBluePlugin.java:336)
E/MethodChannel#plugins.pauldemarco.com/flutter_blue/methods(24348): 	at io.flutter.plugin.common.MethodChannel$IncomingMethodCallHandler.onMessage(MethodChannel.java:233)
E/MethodChannel#plugins.pauldemarco.com/flutter_blue/methods(24348): 	at io.flutter.embedding.engine.dart.DartMessenger.handleMessageFromDart(DartMessenger.java:84)
E/MethodChannel#plugins.pauldemarco.com/flutter_blue/methods(24348): 	at io.flutter.embedding.engine.FlutterJNI.handlePlatformMessage(FlutterJNI.java:865)
E/MethodChannel#plugins.pauldemarco.com/flutter_blue/methods(24348): 	at android.os.MessageQueue.nativePollOnce(Native Method)
E/MethodChannel#plugins.pauldemarco.com/flutter_blue/methods(24348): 	at android.os.MessageQueue.next(MessageQueue.java:379)
E/MethodChannel#plugins.pauldemarco.com/flutter_blue/methods(24348): 	at android.os.Looper.loop(Looper.java:144)
E/MethodChannel#plugins.pauldemarco.com/flutter_blue/methods(24348): 	at android.app.ActivityThread.main(ActivityThread.java:7529)
E/MethodChannel#plugins.pauldemarco.com/flutter_blue/methods(24348): 	at java.lang.reflect.Method.invoke(Native Method)
E/MethodChannel#plugins.pauldemarco.com/flutter_blue/methods(24348): 	at com.android.internal.os.Zygote$MethodAndArgsCaller.run(Zygote.java:245)
E/MethodChannel#plugins.pauldemarco.com/flutter_blue/methods(24348): 	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:921)
E/DartMessenger(24348): Uncaught exception in binary message listener
E/DartMessenger(24348): java.lang.IllegalStateException: Reply already submitted
E/DartMessenger(24348): 	at io.flutter.embedding.engine.dart.DartMessenger$Reply.reply(DartMessenger.java:164)
E/DartMessenger(24348): 	at io.flutter.plugin.common.MethodChannel$IncomingMethodCallHandler.onMessage(MethodChannel.java:253)
E/DartMessenger(24348): 	at io.flutter.embedding.engine.dart.DartMessenger.handleMessageFromDart(DartMessenger.java:84)
E/DartMessenger(24348): 	at io.flutter.embedding.engine.FlutterJNI.handlePlatformMessage(FlutterJNI.java:865)
E/DartMessenger(24348): 	at android.os.MessageQueue.nativePollOnce(Native Method)
E/DartMessenger(24348): 	at android.os.MessageQueue.next(MessageQueue.java:379)
E/DartMessenger(24348): 	at android.os.Looper.loop(Looper.java:144)
E/DartMessenger(24348): 	at android.app.ActivityThread.main(ActivityThread.java:7529)
E/DartMessenger(24348): 	at java.lang.reflect.Method.invoke(Native Method)
E/DartMessenger(24348): 	at com.android.internal.os.Zygote$MethodAndArgsCaller.run(Zygote.java:245)
E/DartMessenger(24348): 	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:921)
D/BluetoothManager(24348): getConnectionState()
D/BluetoothManager(24348): getConnectedDevices
D/BluetoothManager(24348): getConnectionState()
D/BluetoothManager(24348): getConnectedDevices
I/flutter (24348):   DEBUG [🐛  21:32:05 FlutterBluePageController]: BluetoothDeviceState.disconnected
```

This issue doesn't happen in version 0.12.0. My testing device is  Android 8.0.0. Could anyone provide any advice on how to deal with it? Thanks in advance!

**Comment by russelltg** [comment]
I beleive this is fixed by #17

**Comment by vin-lz** [comment]
> I believe this is fixed by #17

It fixes the issue. Thank you very much!
